src/eZ/ContentView/QueryResultsInjector.php

```diff
@@ -108,21 +122,37 @@ private function buildResults(FilterViewParametersEvent $event): iterable
             $pageParam = sprintf('%s_page', $fieldDefinitionIdentifier);
             $page = isset($request) ? $request->get($pageParam, 1) : 1;
 
-            $pager = new Pagerfanta(
-                new QueryResultsPagerFantaAdapter(
-                    $this->queryFieldService, $content, $fieldDefinitionIdentifier
-                )
+            $adapter = new QueryResultsPagerFantaAdapter(
```

Comment:
Maybe it could be better to close it in some factory, what do you think?

Reply:
Why not.

Reply:
I'm not so sure about that.

Introducing a factory here means increasing the complexity of this particular code, expecting different `PagerFantaAdapter`s to be introduced in some cases. Currently added code simply decorates the existing adapter if `debug` property is set to `true`.

It would also not get rid of the `debug` property, since the second, pagination-less variant just a few lines below would also still be required. This would mean it would cause the mentioned factory to be also responsible for that second variant / another service be introduced / `debug` property be left as-is in the injector.
All-in-all, I don't really see the benefit of introducing a factory now. If new adapter variants show up, then I'd be up for it.

Reply:
It is not as complex code that factory is must-have here but I see some advantages of that approach.  It may increase readability - but it is my subjective opinion ;) however similar effect you can achieve by extracting it to a private function. 
I not sure about increasing complexity here. if it really happened, it wouldn't be a problem because current complexity is rather low. However, in my opinion, it should reduce complexity because some logic will be moved out of the class and it should be noticeable when you would decide to write a unit test for that.

but probably it is not a game-changer here ;) and this discussion going to have more line than code ;) it was just proposal

Reply:
Yes, testing where having a factory with an easily describable behavior will make a difference.

Reply:
@kaff @bdunogier code has been redesigned. Instead of changing the behavior of `QueryResultsInjector` and wrapping `PagerFantaAdapter` with a decorator, a production-only service (based on `kernel.debug` parameter) is added to service container that decorates `QueryFieldService` service.